src/odemis/acq/stream/_sync.py

```diff
@@ -774,6 +779,149 @@ def _stopLeeches(self):
+
+        # Activate _updateImage thread
+        self._shouldUpdateImage()
+
+    def _assembleFinalData(self, n, data, pol_idx, pos_polarizations, tot_num, streams):
+        '''
+        :param n: (int) number of the current stream
+        :param data: all acquired data of the streams
+        :param pol_idx: (int) polarisation index related to name as defined in pos_polarizations variable
+        :param pos_polarizations: (list of str) number of possible polarization possibilities
+        :param tot_num: (int) Total number of images to acquire.
+        '''
+        if type(data[0]) != list:
+            logging.error("incorrect definition of _live_data")
+        if n > tot_num:
+            logging.error("wrong value for n, more images taken than totally defined")
+        self._im_needs_recompute.clear()
+
+        #TODO K.K. replace "or" with "and" while also make this part suitable for both polararized as non pol use
+        # problem is pol_idx=0 without polarizing maybe should be pol_idx+1. Result should be better check while stillw working
+        if pol_idx == len(pos_polarizations) - 1 or len(pos_polarizations) == 1:
+            if n == tot_num:
+                data[0] = model.DataArray(
+                                     numpy.mean(data[0], axis=0).astype(data[0][0].dtype),
+                                     data[0][0].metadata.copy())
+                self._raw = data
+                self._liveActivation = True
+        else:
+            raise ValueError('should not have come into this statement error due to: \n pol_idx=%s != len(pos_polarizations)=%s - 1 or n=%s != tot_num=%s \n'
+                             %( pol_idx, len(pos_polarizations), n, tot_num))
+
+        # Update last image and wait for completion
+        self._shouldUpdateImage()
+        # TODO K.K. ERIC include actual wait on finishing of thread instead of sleep?
+        while self._liveActivation:
+            time.sleep(0.05)
+
+    def _progressPixels(self, rgbim):
+        #TODO K.K. ERIC how to easily get write permission
+        # Clear unscanned pixels plus get write permission
+        rgbim = rgbim * numpy.repeat( numpy.expand_dims(self._acq_mask, axis = 2 ), 3 ,axis = 2)
+
+        #TODO K.K. ERIC Get exact RGB values from Eric and fix transparant background or transparant image force to 0.5 value
+        # meaning either fix tranparancy to certain values or preset layover transparacy of SEM and CCD
+        # rgbim = numpy.concatenate((rgbim, numpy.expand_dims(self._acq_mask, axis = 2 )), axis=2)
+        # does not work
+
+        #Blue background
+        rgbim[numpy.invert(self._acq_mask),0] = 0
+        rgbim[numpy.invert(self._acq_mask), 1] = 121
+        rgbim[numpy.invert(self._acq_mask), 2] = 192
+
+        if hasattr(self, "_npixels2scan"):
+            size_preview = self._npixels2scan
+        else:
+            size_preview = 1
+
+        # Orange progress pixels
+        idx = numpy.where(self._acq_mask == False)
```

Comment:
I don't think this is very reliable. In particular, I think in case of "fuzzy", this is not going to work. An alternative which might work could to have a _scanning_area which is basically a left/top/rigth/bottom (either as 4 int, or a numpy slice), which would directly fit on the SEM data. It would be computed/updated by the runAcquisition function.